File: unit-3/sprint-2/day-2/assignments/Perform background task using JobIntentService/app/src/main/java/com/example/performbackgroundtaskusingjobintentservice/MainActivity.java
package com.example.performbackgroundtaskusingjobintentservice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    private Button btntask1, btntask2;
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initviews();
    }

    private void initviews() {
        btntask1 = findViewById(R.id.task1);
        btntask2 = findViewById(R.id.task2);
        intent = new Intent(MainActivity.this, JobIntentService.class);

        btntask1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent.putExtra("Job", "1");
                JobIntentService.ework(MainActivity.this, intent);


            }
        });

        btntask2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                intent.putExtra("Job", "2");
                JobIntentService.ework(MainActivity.this, intent);
            }
        });
    }
}

File: unit-3/sprint-3/day-2/assignments/POST Request/app/src/main/java/com/example/postrequest/MainActivity.kt
package com.example.postrequest

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class MainActivity : AppCompatActivity() {

    private var  responseModel = ResponseModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnPostRequest.setOnClickListener {
            sendData();
        }
    }

    private fun sendData() {
        val email = etEmail.text.toString()
        responseModel.setEmail(email)
        val password = etPassword.text.toString()
        responseModel.setPassword(password)
        val title = etTitle.text.toString()
        responseModel.setTitle(title)
        val requestType = etRequestType.text.toString()
        responseModel.setRequestType(requestType)

        val apiService = Network.getRetrofitInstance().create(
            ApiService::class.java
        )
        val call: Call<ResponseModel?>? = apiService.sendData(responseModel)
        if (call != null) {
            call.enqueue(object : Callback<ResponseModel?> {
                override fun onResponse(call: Call<ResponseModel?>?, response: Response<ResponseModel?>?) {
                    val responseModel: ResponseModel? = response?.body()
                    if (response != null) {
                        val intent = Intent(this@MainActivity, RecieveActivity::class.java)
                        intent.putExtra("email", responseModel?.getEmail())
                        intent.putExtra("password", responseModel?.getPassword())
                        intent.putExtra("title", responseModel?.getTitle())
                        intent.putExtra("requestType", responseModel?.getRequestType())
                        startActivity(intent)
                    }
                }

                override fun onFailure(call: Call<ResponseModel?>?, t: Throwable?) {}
            })
        }
    }
}